### PR TITLE
Limit the maximum number of search vectors to generate per index

### PR DESCRIPTION
**Pull request by NyanKiyoshi**
This fixes a crash when multiple thousand values are to be indexed due to PostgreSQL rejecting the statement. Such crash would look like this:
```
django.db.utils.OperationalError: stack depth limit exceeded
HINT:  Increase the configuration parameter "max_stack_depth" (currently 2048kB), after ensuring the platform's stack depth limit is adequate.
```

Requires https://github.com/saleor/saleor/pull/10261
Task: https://app.clickup.com/t/2549495/CLOUD-1823

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [x] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [x] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [CLOUD-1823 Limit maximum number of search vector operations in SQL statements](https://app.clickup.com/t/2549495/CLOUD-1823)

File: saleor/settings.py [hunk 1]
```diff
@@ -740,6 +740,20 @@ def SENTRY_INIT(dsn: str, sentry_opts: dict):
...
+
+# Maximum related objects that can be indexed in an order
+SEARCH_ORDERS_MAX_INDEXED_PAYMENTS = 20
+SEARCH_ORDERS_MAX_INDEXED_DISCOUNTS = 20
+SEARCH_ORDERS_MAX_INDEXED_LINES = 100
+
+# Maximum related objects that can be indexed in a product
+PRODUCT_MAX_INDEXED_ATTRIBUTES = 100
+PRODUCT_MAX_INDEXED_ATTRIBUTE_VALUES = 100
+PRODUCT_MAX_INDEXED_VARIANTS = 100
```
Comment (patrys) [inline saleor/settings.py:755]: Those are quite low, I think it's not an edge case to have 100 attributes assigned to a complex product, same with 100 variants for something like a multi-choice PC configurator. Can we bump them to 1000?

Comment (NyanKiyoshi) [reply saleor/settings.py:755]: Like so, right?
```suggestion
SEARCH_ORDERS_MAX_INDEXED_PAYMENTS = 20
SEARCH_ORDERS_MAX_INDEXED_DISCOUNTS = 20
SEARCH_ORDERS_MAX_INDEXED_LINES = 100

# Maximum related objects that can be indexed in a product
PRODUCT_MAX_INDEXED_ATTRIBUTES = 1000
PRODUCT_MAX_INDEXED_ATTRIBUTE_VALUES = 100
PRODUCT_MAX_INDEXED_VARIANTS = 1000
```

**Review by maarcingebala — APPROVED** [review]

**Review by korycins — APPROVED** [review]

File: saleor/settings.py (hunk 1 above)
Comment (patrys) [reply saleor/settings.py]: Exactly, yes.

**Pull request merged**